src/ripple/app/hook/impl/applyHook.cpp

```diff
@@ -0,0 +1,4976 @@
+
+            ++ownerCount;
+            XRPAmount const newReserve{
+                view.fees().accountReserve(ownerCount)};
+
+            if (STAmount((*sleAccount)[sfBalance]).xrp() < newReserve)
+                return tecINSUFFICIENT_RESERVE;
+
+
+            adjustOwnerCount(view, sleAccount, 1, j);
+
+        }
+
+        // update state count
+        sleAccount->setFieldU32(sfHookStateCount, stateCount);
+        view.update(sleAccount);
+
+        // create an entry
+        hookState = std::make_shared<SLE>(hookStateKeylet);
+    
+    }
+    
+
+    hookState->setFieldVL(sfHookStateData, data);
+    hookState->setFieldH256(sfHookStateKey, key);
+
+    if (createNew)
+    {
+        bool nsExists = !!view.peek(hookStateDirKeylet);
+
+        auto const page = view.dirInsert(
+            hookStateDirKeylet,
+            hookStateKeylet.key,
+            describeOwnerDir(acc));
+        if (!page)
+            return tecDIR_FULL;
+
+        hookState->setFieldU64(sfOwnerNode, *page);
+
+        // add new data to ledger
+        view.insert(hookState);
+
+        // update namespace vector where necessary
+        if (!nsExists)
+        {
+            if (!sleAccount->isFieldPresent(sfHookNamespaces))
+            {
+                sleAccount->setFieldV256(sfHookNamespaces, STVector256{std::vector<uint256>{ns}});
+                view.update(sleAccount);
+            }
+            else
+            {
+                STVector256 const& vec = sleAccount->getFieldV256(sfHookNamespaces);
+                std::vector<uint256> nv { vec.value() };
+                nv.push_back(ns);
+                sleAccount->setFieldV256(sfHookNamespaces, STVector256{std::move(nv)});
+                view.update(sleAccount);
+            }
+        }
+
+    }
+    else
+    {
+        view.update(hookState);
+    }
+
+    return tesSUCCESS;
+}
+
+hook::HookResult
+hook::apply(
+    ripple::uint256 const& hookSetTxnID, /* this is the txid of the sethook, used for caching (one day) */
+    ripple::uint256 const& hookHash,     /* hash of the actual hook byte code, used for metadata */
+    ripple::uint256 const& hookNamespace,
+    ripple::Blob const& wasm,
+    std::map<
+        std::vector<uint8_t>,          /* param name  */
+        std::vector<uint8_t>           /* param value */
+        > const& hookParams,
+    std::map<
+        ripple::uint256,          /* hook hash */
+        std::map<
+            std::vector<uint8_t>,
+            std::vector<uint8_t>
+        >> const& hookParamOverrides,
+    HookStateMap& stateMap,
+    ApplyContext& applyCtx,
+    ripple::AccountID const& account,     /* the account the hook is INSTALLED ON not always the otxn account */
+    bool hasCallback,
+    bool isCallback,
+    bool isStrong,
+    uint32_t wasmParam,
+    int32_t hookChainPosition,
+    std::shared_ptr<STObject const> const& provisionalMeta)
+{
+
+    HookContext hookCtx =
+    {
+        .applyCtx = applyCtx,
```

Comment:
This surprised me actually. It surprised me more that my compiler let me use them. Possibly quirks mode to stop C developers pulling their hair out. C has had them since 1999